Mathlib/Geometry/Manifold/IntegralCurve.lean

```diff
@@ -132,6 +132,26 @@ lemma IsIntegralCurveAt.isIntegralCurveOn (h : ∀ t ∈ s, IsIntegralCurveAt γ
   obtain ⟨s, hs, h⟩ := isIntegralCurveAt_iff.mp (h t ht)
   exact h t (mem_of_mem_nhds hs)
 
+lemma IsIntegralCurveOn.congr_of_eqOn (hs : IsOpen s) (h : IsIntegralCurveOn γ v s)
+    (hγ : EqOn γ γ' s) : IsIntegralCurveOn γ' v s := by
+  intros t ht
+  rw [← hγ ht]
+  apply (h t ht).congr_of_eventuallyEq
+  exact Filter.eventuallyEq_of_mem (hs.mem_nhds ht) hγ.symm
+
+lemma IsIntegralCurveAt.congr_of_eventuallyEq (h : IsIntegralCurveAt γ v t₀)
+    (hγ : γ =ᶠ[nhds t₀] γ') : IsIntegralCurveAt γ' v t₀ := by
+  obtain ⟨s, hdrv, hs1, hs2⟩ := eventually_nhds_iff.mp (h.and hγ)
+  refine eventually_nhds_iff.mpr ⟨s, fun t ht ↦ ?_, hs1, hs2⟩
+  apply ((hdrv t ht).2 ▸ (hdrv t ht).1).congr_of_eventuallyEq
+  rw [Filter.eventuallyEq_iff_exists_mem]
+  exact ⟨s, hs1.mem_nhds ht, fun t' ht' ↦ (hdrv t' ht').2.symm⟩
+
+lemma IsIntegralCurve.congr (h : IsIntegralCurve γ v) (hγ : γ = γ') :
+    IsIntegralCurve γ' v := by
+  rw [isIntegralCurve_iff_isIntegralCurveOn] at *
+  apply h.congr_of_eqOn isOpen_univ <| (eqOn_univ γ γ').mpr hγ
+
```

Comment:
```suggestion
```
I don't think you want this lemma. For emphasis, note that the proof can be just:
```lean
example (h : IsIntegralCurve γ v) (hγ : γ = γ') : IsIntegralCurve γ' v := hγ ▸ h
```